context.go

```diff
@@ -213,6 +214,7 @@ func (ctx *context) setValueForKey(key string, value interface{}) error {
 		return fmt.Errorf("Cannot set nil as value.")
 	}
 
+	ctx.counters.stores++
```

Comment:
You're increase this variable here, but probably is reading in other coprocess. For this case, it's necessary protect it with the mutex.

Reply:
Actually, it cannot/should not be accessed concurrently.

- `counters.stores` can only be incremented by the goroutine that creates the context, which is the partition-goroutine.
- promise-goroutines (coming from emit confirmations) or the partition-goroutine may read `stores` if they call `ctx.commit()`
- but only one of them will call it (the last one calling `tryCommit()`) and that happens after at least one lock-protected critical section, which will act as a memory barrier anyways.

The comment made me realize though, that `counters.calls` doesn't need a lock too. So I removed that lock and also renamed the counters.